class.jetpack-client-server.php

```diff
@@ -178,9 +156,9 @@ function get_token( $data ) {
 			'code' => $data['code'],
 			'redirect_uri' => add_query_arg( array(
 				'action' => 'authorize',
-				'_wpnonce' => wp_create_nonce( "jetpack-authorize_{$role}_{$redirect}" ),
+				'_wpnonce' => $nonce,
```

Comment:
Nice - this probably fixes an unrelated (and very rare) `redirect_uri` mismatch bug.